### PR TITLE
Fixes/add missing hooks

### PR DESCRIPTION
**Pull request by adamsilverstein**
This pull request adds the following hooks, see https://github.com/WP-API/WP-API/issues/1571:

* attachments: 
rest_insert_attachment (insert, update)
*note: no delete action available

* comments:
rest_insert_comment (insert, update)
rest_delete_comment

* meta:
rest_insert_meta (insert, update)
rest_delete_meta

* posts:
rest_delete_post 

* terms:
rest_insert_term
rest_remove_term

* revisions:
rest_delete_revision

* users:
rest_delete_user

### CODE REVIEW

**Comment by rachelbaker** [comment]
@adamsilverstein Looks like several unit tests break from this PR: https://travis-ci.org/WP-API/WP-API/jobs/81328312

**Comment by adamsilverstein** [comment]
@rachelbaker had some mismatched variable names when I copied in some hooks. Worth a close review, tests should pass now anyway :) Got them running locally as well so I can test before pushing. I'll add some hook tests if I can get to it!

File: tests/test-rest-comments-controller.php [hunk 1]
```diff
@@ -21,6 +21,9 @@ class WP_Test_REST_Comments_Controller extends WP_Test_REST_Controller_Testcase
 	public function setUp() {
 		parent::setUp();
 
+		// Don't fail on flood.
+		remove_filter( 'check_comment_flood', 'check_comment_flood_db' );
```
Comment (danielbachhuber) [inline tests/test-rest-comments-controller.php]: This shouldn't be added as a part of this pull request.

Comment (danielbachhuber) [inline tests/test-rest-comments-controller.php]: > This shouldn't be added as a part of this pull request.

To clarify, I don't think this is the appropriate fix. We should likely resolve #1660, #790

**Comment by joehoyle** [comment]
@adamsilverstein could you revert commit 6053965ae42baf219499450ee352f2a406c8e93b, this should have been fixed in `develop` if you merge that into your branch too. We should then merge this ASAP to stop it going stale.

**Comment by adamsilverstein** [comment]
@joehoyle I will work on cleaning up this PR and reverting commit 6053965.

**Comment by adamsilverstein** [comment]
@joehoyle I reverted 6053965 and merged in develop to ensure a clean diff.

Please let me know if you see any outstanding issues.

**Comment by rachelbaker** [comment]
@adamsilverstein Thank you!

**Comment by adamsilverstein** [comment]
Woohoo! thanks!

**Pull request merged**